example_package/README.md

```diff
@@ -0,0 +1,456 @@
+Wczytuje liczbę `a` (int, float, ...) która ma być w podanym zakresie od `b` do `c`.
+
+Podawanie zakresu może wydawać się żmudne, ale pozwala zapobiec, że ktoś poda nieskończenie długie słowo, albo że przegapimy sprawdzenie czy liczba jest w odpowiednim zakresie.
+
+Wszystkie te funkcje w przypadku gdy wczytają coś, co nie pasuje do opisu, zgłoszą błąd.
+
+#### **CheckerVerdict**
+
+oi.h udostępnia nam obiekt `checker_verdict` klasy CheckerVerdict.
+Używamy go standardowo `oi::checker_verdict.[coś]`.
+Udostępnia on nam poniższe funkcje:
+
+- **exit_ok()** -
+Kończy sprawdzanie z sukcesem dając 100 punktów.
+Zwraca `OK\n\n100\n`.
+- **exit_ok_with_score(int score, Msg&&... msg)** -
+Kończy sprawdzanie z sukcesem z podanym wynikiem i wiadomością/ciami.
+Zwraca `OK\n[msg]\n[score]\n`.
+- **set_partial_score(int score, Msg&&... msg)** -
+Ustawia wynik częściowy który zostanie zwrócony gdy nastąpi błąd.
+Czyli zamiast 0 punktów otrzyma się tyle ile się przypisało z danym komentarzem.
+- **exit_wrong(Msg&&... msg)** -
+Kończ sprawdzanie z błędem i daje 0 punktów, chyba, że ustawiono partial_score.
+Zwraca `WRONG\n[msg]\n0\n` albo `OK\n[partial_score_msg]; [msg]\n[partial_score]\n` albo `OK\n[msg]\n[partial_score]\n`.
+
+Jeżeli w programie używamy `partial_score` to:
+
+- Trzymajmy zmienną mówiącą jaki jest wynik częściowy.
+- Trzymajmy zmienną mówiącą ile user może zdobyć punktów (do_zdobycia).
+- Używamy `exit_ok_with_score(do_zdobycia)` zamiast `exit_ok()`.
+
+#### **checker_test**
+
+oi.h udostępnia możliwość pisania testów do chekerki.
+Te testy są uruchamiane tylko lokalnie.
+Istnieją 2 sposoby ich pisania.
+Przykład obu z nich jest zaimplementowany w przykładowym `abcchk.cpp`.
+
+#### **InwerVerdict**
+
+oi.h udostępnia nam obiekt `inwer_verdict` klasy InwerVerdict.
+Używamy go jako strumień wyjścia, a mianowicie:
+`oi::inwer_verdict.[coś] << [msg]`.
+Gdzie `msg` to wiadomość którą chcemy pokazać przed zakończeniem.
+Natomiast `coś` to jedna z podanych opcji:
+
+- **exit_ok()** - Kończy program pomyślnie.
+- **exit_wrong()** - Kończy program z błędem.
+
+My będziemy używać tylko `oi::inwer_verdict.exit_ok() << [msg]`.
+Druga opcja jest używana systemowo i będziemy ją zgłaszać np. przez `oi_assert()` lub `oi::bug(Msg&&... msg)`.
+
+#### **bug**
+
+Wywołując `oi::bug(Msg&&... msg)`, program zakończy się niepowodzeniem.
+Wyświetli on wtedy podaną wiadomość/ci, w formacie `BUG: [msg]` i zakończy działanie z kodem 2.
+
+#### **oi_assert**
+
+Działa podobnie do zwykłego asserta.
+Wywołując `oi_assert(condition, msg...);`, sprawdzi założenie, a jak będzie błędne to poda dokładny komunikat co jest nie tak.
+Wypisze on `[FILE]:[LINE]: [func]: Assertion '[condition]' failed.` lub
+`[FILE]:[LINE]: [func]: Assertion '[condition]' failed: [msg]`
+
+#### **Random**
+
+Służy do losowania wartości i jest wymagane używać go zamiast zwykłego rand() lub std::mt19937 lub innych mechanizmów losujących.
+Zapewnia on, że liczby są rzeczywiście (pseudo) losowe.
```

Comment:
Jakoś wspomnieć jeszcze, że jest to "standardowy" sposób, bo mt19937 też będzie zawsze "dobre"

Reply:
nie wiem jak to wpleść

Reply:
Zapewnia on uniwersalny sposób generowania liczb (pseudo) losowych. ?